shellguide.md

```diff
@@ -1006,8 +1006,8 @@ echo $(( hr * 3600 + min * 60 + sec )) # prints 7530 as expected
 
 Lower-case, with underscores to separate words. Separate libraries with
 `::`. Parentheses are required after the function name. The
-keyword `function` is optional, but must be used consistently
-throughout a project.
+keyword `function` is optional but not POSIX. Consistency throughout a project
```

Comment:
IMO I'd rather this be a footnote, maybe something like: ``[^posix]: Note that POSIX requires the `function` keyword; if your project requires POSIX-compatibility the `function` keyword will be needed.`` I feel it detracts from the point to call it out inline (since anyone requiring POSIX-compatibility doesn't need this advice to know that the keyword is necessary for their script to parse correctly).

Reply:
i think you have this backwards: `function` is *not* in POSIX, so a POSIX-compliant script would never use the keyword.

i think keeping this section as-is is OK (since it only talks about consistency), and leaving the POSIX bit to the section below.

Reply:
Yes, I misread the added text to mean "required" not "unusable" (so there's some wordsmithing to do here regardless). I'm OK with leaving the text as-is.